pandaserver/dataservice/adder_atlas_plugin.py

```diff
@@ -237,324 +235,321 @@ def update_outputs(self):
-                                    if (tmp_src_ddm != tmp_dst_ddm) or (tmp_src_ddm == tmp_dst_ddm and file.destinationDBlockToken.count(",") != 0):
-                                        opt_sub = {
-                                            "DATASET_COMPLETE_EVENT": [
-                                                f"http://{panda_config.pserverhosthttp}:{panda_config.pserverporthttp}/server/panda/datasetCompleted"
-                                            ]
-                                        }
-                                        # append
-                                        if file_destination_dispatch_block not in sub_map:
-                                            sub_map[file_destination_dispatch_block] = []
-                                            # sources
-                                            opt_source = {}
-                                            # set sources
-                                            if file.destinationDBlockToken in [
-                                                "NULL",
-                                                "",
-                                            ]:
-                                                # use default DDM ID as source
-                                                opt_source[tmp_src_ddm] = {"policy": 0}
-                                            else:
-                                                # convert token to DDM ID
-                                                ddm_id = tmp_src_ddm
-                                                # use the first token's location as source for T1D1
-                                                tmp_src_token = file.destinationDBlockToken.split(",")[0]
-                                                if tmp_src_token in src_site_spec.setokens_output[scope_src_site_spec_output]:
-                                                    ddm_id = src_site_spec.setokens_output[scope_src_site_spec_output][tmp_src_token]
+                                            # convert token to DDM ID
+                                            ddm_id = tmp_src_ddm
+                                            # use the first token's location as source for T1D1
+                                            tmp_src_token = file.destinationDBlockToken.split(",")[0]
+                                            if tmp_src_token in src_site_spec.setokens_output[scope_src_site_spec_output]:
+                                                ddm_id = src_site_spec.setokens_output[scope_src_site_spec_output][tmp_src_token]
+                                            opt_source[ddm_id] = {"policy": 0}
+                                        # T1 used as T2
+                                        if (
+                                            src_site_spec.cloud != self.job.cloud
+                                            and (not tmp_src_ddm.endswith("PRODDISK"))
```

Comment:
Did you check in CRIC if there are RSEs that end in PRODDISK or of ATLASPRODDISK type? I think we had been removing PRODDISK code in the previous exercises.

Reply:
There are a few more PRODDISK things around the file